Karnety

ZuberRS03 opened the pull request:
Gotowa encja o karnetach, doszedłem do wniosku że łatwiej i bardziej rozwojowa jest opcja stworzenia osobnej encji niż wplatanie tego w wypożyczenia.
Poniżej lista zapytań do testowania.
```xml
<folderState>
                <option name="id" value="ce69de65-24da-4b16-be34-c23211d18ce8" />
                <option name="name" value="Pass" />
                <option name="requests">
                  <list>
                    <requestState>
                      <option name="description" value="Pobierz wszystkie dostępne karnety" />
                      <option name="id" value="e357fdaa-30b5-40de-adb3-1c33824df8e6" />
                      <option name="method" value="GET" />
                      <option name="name" value="All passes" />
                      <option name="sortWeight" value="1000000" />
                      <option name="url" value="http://localhost:8080/api/v1/pass/all" />
                    </requestState>
                    <requestState>
                      <option name="description" value="Pobierz wszystkie karnety" />
                      <option name="id" value="ea3f7251-ba83-4e86-88c5-378b588e0855" />
                      <option name="method" value="GET" />
                      <option name="name" value="All passes for Admin" />
                      <option name="sortWeight" value="2000000" />
                      <option name="url" value="http://localhost:8080/api/v1/authorized/admin/pass/all" />
                    </requestState>
                    <requestState>
                      <option name="description" value="Pobierz konkretny karnet" />
                      <option name="id" value="9855fa51-7c98-40e5-bf50-dadc45a053ce" />
                      <option name="method" value="GET" />
                      <option name="name" value="One pass" />
                      <option name="sortWeight" value="3000000" />
                      <option name="url" value="http://localhost:8080/api/v1/pass/1" />
                    </requestState>
                    <requestState>
                      <option name="description" value="Pobierz wszystkie karnety usera" />
                      <option name="id" value="4226cf20-1ffc-40d0-8321-738e68cd2724" />
                      <option name="method" value="GET" />
                      <option name="name" value="User passes" />
                      <option name="sortWeight" value="4000000" />
                      <option name="url" value="http://localhost:8080/api/v1/authorized/pass/passHistory" />
                    </requestState>
                    <requestState>
                      <option name="description" value="Pobierz wszystkie karnety" />
                      <option name="id" value="b21d59e8-2a24-4440-a898-a370a137bc25" />
                      <option name="method" value="GET" />
                      <option name="name" value="All users passes for Admin" />
                      <option name="sortWeight" value="5000000" />
                      <option name="url" value="http://localhost:8080/api/v1/authorized/admin/pass/passHistory/all" />
                    </requestState>
                    <requestState>
                      <option name="description" value="Kup jeden karnet" />
                      <option name="id" value="747769e0-7752-49ea-bf17-1951d6e4659c" />
                      <option name="method" value="POST" />
                      <option name="name" value="Buy new pass" />
                      <option name="sortWeight" value="6000000" />
                      <option name="url" value="http://localhost:8080/api/v1/authorized/pass/buyNew/3" />
                    </requestState>
                    <requestState>
                      <option name="body">
                        <bodyState>
                          <option name="raw" value="{&#10;  &quot;name&quot;: &quot;karnet 30 dniowy&quot;,&#10;  &quot;description&quot;: &quot;karnet na 30 dni z nielimitowanymi wejściami&quot;,&#10;  &quot;price&quot;: 150.99,&#10;  &quot;days&quot;: 30,&#10;  &quot;status&quot;: &quot;VISIBLE&quot;&#10;}" />
                          <option name="type" value="JSON" />
                        </bodyState>
                      </option>
                      <option name="description" value="Dodaj karnet do kupna" />
                      <option name="id" value="612399a4-6f6f-4818-8704-d4d4eddb4d61" />
                      <option name="method" value="POST" />
                      <option name="name" value="Add new pass" />
                      <option name="sortWeight" value="7000000" />
                      <option name="url" value="http://localhost:8080/api/v1/authorized/admin/pass/add" />
                    </requestState>
                    <requestState>
                      <option name="description" value="Usuń karnet" />
                      <option name="id" value="d162d0b2-5e93-4083-ba7d-3097c8245334" />
                      <option name="method" value="DELETE" />
                      <option name="name" value="Delete pass" />
                      <option name="sortWeight" value="8000000" />
                      <option name="url" value="http://localhost:8080/api/v1/authorized/admin/pass/delete/3" />
                    </requestState>
                    <requestState>
                      <option name="description" value="Ukrywa karnet dla userów" />
                      <option name="id" value="862ce1ef-201f-4463-887e-15621b4f8947" />
                      <option name="method" value="PUT" />
                      <option name="name" value="Disable pass" />
                      <option name="sortWeight" value="9000000" />
                      <option name="url" value="http://localhost:8080/api/v1/authorized/admin/pass/disable/2" />
                    </requestState>
                    <requestState>
                      <option name="body">
                        <bodyState>
                          <option name="raw" value="{&#10;  &quot;name&quot;: &quot;karnet 20 dniowy&quot;,&#10;  &quot;description&quot;: &quot;karnet na 20 dni z nielimitowanymi wejściami&quot;,&#10;  &quot;price&quot;: 150.99,&#10;  &quot;days&quot;: 20,&#10;  &quot;status&quot;: &quot;HIDDEN&quot;&#10;}" />
                          <option name="type" value="JSON" />
                        </bodyState>
                      </option>
                      <option name="description" value="Zaaktualizuj karnet o danym id" />
                      <option name="id" value="d16d4b8d-7a70-49be-8adb-e0d0f127d2b3" />
                      <option name="method" value="PUT" />
                      <option name="name" value="Update pass" />
                      <option name="sortWeight" value="10000000" />
                      <option name="url" value="http://localhost:8080/api/v1/authorized/admin/pass/update/1" />
                    </requestState>
                  </list>
                </option>
                <option name="sortWeight" value="5000000" />
              </folderState>
```

ZuberRS03 committed:
Pierwsza działająca wersja encji o karnetach

Jeszcze nie testowana dogłębnie. Sprawdziłem tylko czy wszystko działa kiedy podam dobre dane, do sprawdzenia są zwrotki przy problemach i zabezpieczenia.

ZuberRS03 committed:
Dodany config i poprawiono kilka błędów

Dodałem plik z dwoma karnetami żeby były od początku.
Poprawiłem kilka niedociągnięć z wyświetlaniem i zwracanymi błędami.
Teraz nie da się kupić ukrytego karnetu.
Przy każdym zapytaniu o karnety użytkownika status karnetów jest sprawdzany i zmieniany jeżeli karnet wygasł.

xEdziu reviewed (changes requested):
![image](https://github.com/user-attachments/assets/f91320d7-9096-4176-96aa-be24252dda64)

Można kupić kilka karnetów - nie ma sprawdzenia czy użytkownik ma już kupiony / ważny karnet

xEdziu commented on src/main/java/pwr/isa/klama/pass/PassController.java:
```diff
@@ -0,0 +1,77 @@
+        return passService.updatePass(pass, passId);
+    }
+
+    //TODO: Dodanie sprawdzenia statusu karnetu przy każdym get i ewentualna zmiana statusu, zmiana na posdtawie daty ważności
```
tutaj todo też do zrobienia :D - ale to robisz w serwisie bardziej niz tutaj

ZuberRS03 replied:
To już w sumie zrobiłem tylko zapomniałem wywalić TODO :)

ZuberRS03 commented:
Wiem o tym ale podszedłem do tego jak MPK we wrocławiu, też jak masz już ważny bilet przypisany do karty to pozwoli Ci kupić kolejny, to problem kupującego że kupi drugi. Ewentualnie moge przerobić że kupno kolejnego przedłuża wartość poprzedniego ale to nie będzie widać całej historii karnetów.

xEdziu commented:
> Wiem o tym ale podszedłem do tego jak MPK we wrocławiu, też jak masz już ważny bilet przypisany do karty to pozwoli Ci kupić kolejny, to problem kupującego że kupi drugi. Ewentualnie moge przerobić że kupno kolejnego przedłuża wartość poprzedniego ale to nie będzie widać całej historii karnetów.

Ah, no dobra - w sumie jak chcesz w takim razie - w założeniach było po prostu inaczej xD

ZuberRS03 commented:
> > Wiem o tym ale podszedłem do tego jak MPK we wrocławiu, też jak masz już ważny bilet przypisany do karty to pozwoli Ci kupić kolejny, to problem kupującego że kupi drugi. Ewentualnie moge przerobić że kupno kolejnego przedłuża wartość poprzedniego ale to nie będzie widać całej historii karnetów.
> 
> Ah, no dobra - w sumie jak chcesz w takim razie - w założeniach było po prostu inaczej xD

Wiesz, jak było w założeniach to moge to dorobić tylko będzie troche głupio jak będzie tak totalnie blokowało kupno karnetu na 30 dni gdy np zostało Ci 1 dzień ważności poprzedniego

xEdziu commented:
No racja racja - zostawiamy tak jak jest

ZuberRS03 committed:
Drobne poprawki

Dodałem sprawdzanie czy nie zmienia się status wypożyczenia na podobnej zasadzie co w karnetach
Usunąłem todo o którym zapomniałem usunąć
Zmieniłem napisy w konfigu na polskie żeby było wszystko spójne

xEdziu reviewed (approved):
Jest super - czy coś zostało stąd do zrobienia?
